src/main/java/gdx/liftoff/ui/panels/CompleteButtonsPanel.java

```diff
@@ -68,23 +69,36 @@ public void populate(boolean fullscreen) {
         table.add(ideaButton);
         addHandListener(ideaButton);
         try {
-            List<String> findIntellijExecutable = (UIUtils.isWindows) ? Arrays.asList("where.exe", "idea") : Arrays.asList("which", "idea");
+            if (UIUtils.isWindows) {
+                List<String> findIntellijExecutable = Arrays.asList("where.exe", "idea");
 
-            Process whereProcess = new ProcessBuilder(findIntellijExecutable).start();
-            if (whereProcess.waitFor() == 0) {
-                intellijPath = new BufferedReader(new InputStreamReader(whereProcess.getInputStream())).readLine();
-                ideaButton.setDisabled(false);
-            } else {
-                if (UIUtils.isLinux) {
-                    intellijPath = findFlatpakIntellij();
-                    intellijIsFlatpak = true;
+                Process whereProcess = new ProcessBuilder(findIntellijExecutable).start();
+
+                if (whereProcess.waitFor() == 0) {
+                    intellijPath = new BufferedReader(new InputStreamReader(whereProcess.getInputStream())).readLine();
                     ideaButton.setDisabled(false);
-                } else if (UIUtils.isWindows) {
+                } else {
                     intellijPath = findManuallyInstalledIntellijOnWindows();
                     ideaButton.setDisabled(false);
-                } else {
-                    throw new Exception("IntelliJ not found");
                 }
+            } else if (UIUtils.isLinux) {
+                List<String> whichCommand = new ArrayList<>();
+                whichCommand.add("which");
+                List<String> possibleIdeaNames = getManuallyInstalledLinuxIdeaNames();
+                whichCommand.addAll(possibleIdeaNames);
+
+                Process whichProcess = new ProcessBuilder(whichCommand).start();
+
+                if (whichProcess.waitFor() < possibleIdeaNames.size()) {
```

Comment:
I have no idea what the return values for `which` are or mean. I would imagine 0 means "returned successfully", but the less-than sign here suggests 1 or 2 would also be valid? Negative return values, which I don't know if they can occur or not, would also be less than size. This section doesn't seem very robust, or maybe I just don't understand how `which` works.

Reply:
The contract of which is as follows:

> Which returns the number of failed arguments, or -1 when no `programname´ was given. 

In this case, we have 3 program names. So any exit code less than `3` is acceptable (we expect that at least one of these names will be found). A `0` exit code means that none of the programnames matched, and in this case, the Flatpak branch will be executed. Therefore, there should be no `-1` exit code. However, a double-check wouldn't hurt.

Reply:
OK, that works -- you meant a 0 exit code means that none failed, right? Meaning the arch command was found, and flatpak won't be needed. The main concern I had here was just that the exit code system used is quite different from other tools (where non-zero returns usually signify an error), and it didn't make sense without some documentation.

Reply:
You're absolutely right. A 0 exit code does indeed mean that none of the checks failed so flatpak won't be needed. I'm glad we've come to a consensus on this!